spec/controllers/omniauth_callbacks_controller_spec.rb

```diff
@@ -195,6 +195,26 @@
       let(:oauth_user) { User.find_by(email: oauth_email) }
       let(:user) { oauth_user }
       before { get :facebook }
+      
+      context 'when third party avatar is not nil' do
+        it 'uploads avatar when third_party_avatar has changed' do
+          new_avatar = 'http://example.com/images/different_profile.jpeg'
+          request.env['omniauth.auth']['info']['image'] = new_avatar
+          get :facebook
```

Comment:
For this failing test. you could debug using `byebug` in this action. Specifically, I would look to check that `request.env['omniauth.auth']['info']['image']` is really set to `new_avatar`.

Might need to stub `request.env['omniauth.auth']['info']['image'] = new_avatar` correctly